internal/api/core/core.go

```diff
@@ -51,6 +51,10 @@ func New(conf config.Config, banner string) (*app.Runner, dependency.Persistence
 		DTS:          persistenceManager.GetDatasource(),
 		GlobalDTS:    persistenceManager.GetGlobalDatasource(),
 		Crypto:       serviceManager.GetCrypto(),
+		RBAC:         serviceManager.GetRBAC(),
+		JWTMiddleware: serviceManager.GetJWT().Middleware(func(echo.Context) bool {
```

Comment:
:/ I was hoping by just generating the middleware and then calling it directly in the proxy that it'd keep those anonymous endpoints, but it seems you're right. It was definitely a cause of confusion for me that the proxy is implemented as a middleware rather than just a regular route so I'm happy to investigate moving it across

Reply:
If you want to rewrite it using the route, I would propose to do it in another PR and then go back to this once the refactoring is done.
By the way if rewriting the proxy using only thé route doesn't work, perhaps a combination with route and middleware would make it.
Like fake route that doesn't do anything but is just there to register the path and then a middleware apply to this route that is actually doing the job

Reply:
https://github.com/perses/perses/pull/1644